### ISSUE TITLE
Form components cannot have <div> inside a <label>

### ISSUE THREAD

**Issue by ernestogutierrez** [Shopify/polaris-react-archive#88]
# Issue summary

The [Example App: CDN Styles](https://github.com/Shopify/polaris/tree/master/examples/cdn-styles) and the [Shopify PolarisComponents page](https://polaris.shopify.com/components/forms/checkbox#navigation) show some forms components where the plain html have a `<div>` element inside of a `<label>` element. 

This is not a W3C standard. If you use the [validator.w3.org](http://validator.w3.org) you can check the error.

Some of the components affected are:

- Checkbox
- ChoiceList
- Radio Button

## Expected behavior

This should be changed so the `<label>` just have the label, text, and not a `<div>`
div> inside.

## Actual behavior

This is an example HTML:
```
 <li>
    <label class="Polaris-Choice" for="Checkbox1">
    <div class="Polaris-Choice__Control">
        <div class="Polaris-Checkbox">
        <input type="checkbox" id="Checkbox1" name="ChoiceList1[]" value="false" class="Polaris-Checkbox__Input" aria-invalid="false">
        <div class="Polaris-Checkbox__Backdrop"></div>
        <div class="Polaris-Checkbox__Icon">
            <span class="Polaris-Icon">
            <svg class="Polaris-Icon__Svg" viewBox="0 0 20 20">
                <g fill-rule="evenodd">
                <path d="M8.315 13.859l-3.182-3.417a.506.506 0 0 1 0-.684l.643-.683a.437.437 0 0 1 .642 0l2.22 2.393 4.942-5.327a.437.437 0 0 1 .643 0l.643.684a.504.504 0 0 1 0 .683l-5.91 6.35a.437.437 0 0 1-.642 0"></path>
                <path d="M8.315 13.859l-3.182-3.417a.506.506 0 0 1 0-.684l.643-.683a.437.437 0 0 1 .642 0l2.22 2.393 4.942-5.327a.437.437 0 0 1 .643 0l.643.684a.504.504 0 0 1 0 .683l-5.91 6.35a.437.437 0 0 1-.642 0"></path>
                </g>
            </svg>
            </span>
        </div>
        </div>
    </div>
    <div class="Polaris-Choice__Label">I accept the Terms of Service</div>
    </label>
</li>
```

This is the error that the W3C validator generates:

```
Error: Element div not allowed as child of element label in this context. (Suppressing further errors from this subtree.)
From line 207, column 29; to line 207, column 63
          <div class="Polaris-Choice__Label">
Contexts in which element div may be used:
Where flow content is expected.
As a child of a dl element.
Content model for element label:
Phrasing content, but with no descendant labelable elements unless it is the element’s labeled control, and no descendant  label elements.
```

Additionally, the Visual Studio editor also complains about that error.

**Comment by lemonmade** [comment]
Right you are, thanks for the issue @ernestogutierrez. I'll add an issue to tackle this on our internal tracker and close this once we've shipped a fix.

**Comment by lemonmade** [comment]
Should be fixed by v1.1.0
